### ISSUE TITLE
DenseNet predict different labels for the same input. 

### ISSUE THREAD

**Issue by springshuai** [pytorch/vision#370]
I generated 5 randn arrays and feed them into the pre-trained DenseNet in different ways. Predictions should be the same no matter the order I feed them in, whereas some things seem pretty wrong here. Can anyone help me with this?

The code is a bit long, but most are just repeating things:
`import torch`
`from torch.autograd import Variable`
`from torch import Tensor`
`from torchvision.models import densenet161`
`import numpy as np`
`densenet_full = densenet161(pretrained = True)`
`torch.manual_seed(2122)`
`sample_0 = torch.Tensor(torch.randn(3, 224, 224)).unsqueeze(0)`
`sample_1 = torch.Tensor(torch.randn(3, 224, 224)).unsqueeze(0)`
`sample_2 = torch.Tensor(torch.randn(3, 224, 224)).unsqueeze(0)`
`sample_3 = torch.Tensor(torch.randn(3, 224, 224)).unsqueeze(0)`
`sample_4 = torch.Tensor(torch.randn(3, 224, 224)).unsqueeze(0)`
`variable_0 = Variable(sample_0)`
`variable_1 = Variable(sample_1)`
`variable_2 = Variable(sample_2)`
`variable_3 = Variable(sample_3)`
`variable_4 = Variable(sample_4)`
`variable_duo = Variable(torch.cat((sample_0, sample_1), dim=0))`
`variable_trio = Variable(torch.cat((sample_0, sample_1, sample_2), dim=0))`
`variable_quartet = Variable(torch.cat((sample_0, sample_1, sample_2, sample_3), dim=0))`
`variable_quintet = Variable(torch.cat((sample_0, sample_1, sample_2, sample_3, sample_4), dim=0))`

`pred_0 = densenet_full.forward(variable_0).data.numpy()`
`pred_1 = densenet_full.forward(variable_1).data.numpy()`
`pred_2 = densenet_full.forward(variable_2).data.numpy()`
`pred_3 = densenet_full.forward(variable_3).data.numpy()`
`pred_4 = densenet_full.forward(variable_4).data.numpy()`

`pred_duo = densenet_full.forward(variable_duo).data.numpy()`
`pred_trio = densenet_full.forward(variable_trio).data.numpy()`
`pred_quartet = densenet_full.forward(variable_quartet).data.numpy()`
`pred_quintet = densenet_full.forward(variable_quintet).data.numpy()`

`print('Prediction for the 0th random number is: {}'.format(np.argmax(pred_0)))`
`print('Prediction for the 1st random number is: {}'.format(np.argmax(pred_1)))`
`print('Prediction for the 2nd random number is: {}'.format(np.argmax(pred_2)))`
`print('Prediction for the 3rd random number is: {}'.format(np.argmax(pred_3)))`
`print('Prediction for the 4th random number is: {}'.format(np.argmax(pred_4)))`

`print('Predictions for the two random numbers are: {} and {} respectively.'
       .format(np.argmax(pred_duo, axis=1)[0], 
              np.argmax(pred_duo, axis=1)[1]))`
`print('Predictions for the three random numbers are: {}, {} and {} respectively.'
      .format(np.argmax(pred_trio, axis=1)[0], 
              np.argmax(pred_trio, axis=1)[1], 
              np.argmax(pred_trio, axis=1)[2]))`
`print('Predictions for the four random numbers are: {}, {}, {} and {} respectively.'
      .format(np.argmax(pred_quartet, axis=1)[0], 
              np.argmax(pred_quartet, axis=1)[1], 
              np.argmax(pred_quartet, axis=1)[2], 
              np.argmax(pred_quartet, axis=1)[3]))`
`print('Predictions for the five random numbers are: {}, {}, {}, {} and {} respectively.'
      .format(np.argmax(pred_quintet, axis=1)[0], 
              np.argmax(pred_quintet, axis=1)[1], 
              np.argmax(pred_quintet, axis=1)[2], 
              np.argmax(pred_quintet, axis=1)[3],
              np.argmax(pred_quintet, axis=1)[4]))`

Following are the results:

Prediction for the 0th random number is: 600
Prediction for the 1st random number is: 600
Prediction for the 2nd random number is: 600
Prediction for the 3rd random number is: 463
Prediction for the 4th random number is: 600
Predictions for the two random numbers are: 600 and 973 respectively.
Predictions for the three random numbers are: 596, 973 and 921 respectively.
Predictions for the four random numbers are: 596, 879, 921 and 899 respectively.
Predictions for the five random numbers are: 600, 879, 921, 899 and 852 respectively.

I.e., predictions are varying depending on whether it is inputted by itself or together with others?

**Comment by yannadani** [comment]
You have to put the model to test mode. Add `densenet_full.eval()` before you do the forward pass.
